### ISSUE TITLE
--with-cuda failes to find libcuda.so

### ISSUE THREAD

**Issue by BKitor** [open-mpi/ompi#12509]
OpenMPI can fail to find libcuda.so and will build without opal acclerator cuda when --with-cuda/--with-cuda-libdir is specified. 
This was already reported as a bug in #12264 and fixed in #12382, but the bug persists. 
I noticed in with a v5.0.3 tarball, and have been able to reproduce on master. 

## Details of the problem

```shell
user@bigtwin1d:~/bkitor/bk_share/ompi_builds/ompi[master]$ ./build/bin/ompi_info | grep 'Configure command'
  Configure command line: '--prefix=/home/user/bkitor/bk_share/ompi_builds/ompi/build' '--with-cuda=/usr/local/cuda' '--with-ofi=/usr/local'
user@bigtwin1d:~/bkitor/bk_share/ompi_builds/ompi[master]$ ./build/bin/ompi_info | grep 'MCA accelerator'
         MCA accelerator: null (MCA v2.1.0, API v1.0.0, Component v5.1.0)
```

```shell
user@bigtwin1d:~/bkitor/bk_share/ompi_builds/ompi[master]$ ./build/bin/ompi_info | grep 'Configure command'
  Configure command line: '--prefix=/home/user/bkitor/bk_share/ompi_builds/ompi/build' '--with-cuda=/usr/local/cuda' '--with-cuda-libdir=/usr/local/cuda' '--with-ofi=/usr/local'
user@bigtwin1d:~/bkitor/bk_share/ompi_builds/ompi[master]$ ./build/bin/ompi_info | grep 'MCA accelerator'
         MCA accelerator: null (MCA v2.1.0, API v1.0.0, Component v5.1.0)
```

The crux of the issue is that `/usr/local/cuda` is a symlink, and the `find` command in opal_check_cuda.m4 won't follow it by default. 
Adding the `-H` flag should fix the issue.

**Comment by jsquyres** [comment]
This is a different issue than was reported by #12264, but still falls under the same end-visible symptom of not being able to find the CUDA libraries.

Thanks for reporting, and thanks for submitting the PR!

**Comment by janjust** [comment]
fixed with #12510

**Comment by janjust** [comment]
https://github.com/open-mpi/ompi/pull/12513 I have the cherry-pick up already
